ci/scripts/build-project.sh

```diff
@@ -5,5 +5,10 @@ source $(dirname $0)/common.sh
 repository=$(pwd)/distribution-repository
 
 pushd git-repo > /dev/null
-./gradlew -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false --no-daemon --max-workers=4 -PdeploymentRepository=${repository} build publishAllPublicationsToDeploymentRepository
+if [[ -d /opt/openjdk-secondary ]]; then
+  secondary_java_version=$( ./$(dirname $0)/get-secondary-java-version.sh )
```

Comment:
It'd be nice if we could pass in the toolchain java version rather than figuring it out using the script (clever though it is). I'm not 100% sure that's possible though due to how pipeline.yml is structured. I can experiment a bit when we come to merging this.

Reply:
I wasn't sure either and that was the only way I could more or less play through (at least in my head). It would be great if I can leave this to you after I've tackled the other remarks.